platforms/core-configuration/base-diagnostics/src/main/java/org/gradle/api/plugins/HelpTasksPlugin.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.diagnostics.ProjectReportTask;
+import org.gradle.api.tasks.diagnostics.PropertyReportTask;
+import org.gradle.api.tasks.diagnostics.TaskReportTask;
+import org.gradle.api.tasks.diagnostics.DiagnosticsTaskNames;
+import org.gradle.configuration.Help;
+
+import static org.gradle.api.internal.project.ProjectHierarchyUtils.getChildProjectsForInternalUse;
+
+/**
+ * Adds various reporting tasks that provide information about the project.
+ *
+ * @see <a href="https://gradle.org/help/">Getting additional help with Gradle</a>
+ */
+public abstract class HelpTasksPlugin implements Plugin<Project>, DiagnosticsTaskNames {
```

Comment:
I am skeptical of implementing `DiagnosticsTaskNames` here. It doesn't really seem necessary given that we expose these properties directly on the plugin

Reply:
Discussed above.